federation/pkg/federation-controller/dns/dns.go

```diff
@@ -175,179 +303,109 @@ func getResolvedEndpoints(endpoints []string) ([]string, error) {
-	//         - OR a CNAME record to the next level up, i.e. mysvc.myns.myfed.svc.r1.mydomain.com  (if a healthy shard does not exist in zone z1)
-	// mysvc.myns.myfed.svc.r1.federation
-	//         - a set of A records to IP addresses of all healthy shards in region r1, if one or more of these exist
-	//         - OR a CNAME record to the next level up, i.e. mysvc.myns.myfed.svc.mydomain.com (if no healthy shards exist in region r1)
-	// mysvc.myns.myfed.svc.federation
-	//         - a set of A records to IP addresses of all healthy shards in all regions, if one or more of these exist.
-	//         - no record (NXRECORD response) if no healthy shards exist in any regions)
-	//
-	// For each cached service, cachedService.lastState tracks the current known state of the service, while cachedService.appliedState contains
-	// the state of the service when we last successfully sync'd it's DNS records.
-	// So this time around we only need to patch that (add new records, remove deleted records, and update changed records.
-	//
-	if s == nil {
-		return fmt.Errorf("nil ServiceController passed to ServiceController.ensureDnsRecords(clusterName: %s, cachedService: %v)", clusterName, cachedService)
-	}
-	if s.dns == nil {
-		return nil
-	}
-	if cachedService == nil {
-		return fmt.Errorf("nil cachedService passed to ServiceController.ensureDnsRecords(clusterName: %s, cachedService: %v)", clusterName, cachedService)
-	}
-	serviceName := cachedService.lastState.Name
-	namespaceName := cachedService.lastState.Namespace
-	zoneNames, regionName, err := s.getClusterZoneNames(clusterName)
-	if err != nil {
+	if err := changeSet.Apply(); err != nil {
 		return err
 	}
-	if zoneNames == nil {
-		return fmt.Errorf("failed to get cluster zone names")
-	}
-	serviceDnsSuffix, err := s.getServiceDnsSuffix()
-	if err != nil {
-		return err
-	}
-	zoneEndpoints, regionEndpoints, globalEndpoints, err := s.getHealthyEndpoints(clusterName, cachedService)
-	if err != nil {
-		return err
-	}
-	commonPrefix := serviceName + "." + namespaceName + "." + s.federationName + ".svc"
-	// dnsNames is the path up the DNS search tree, starting at the leaf
-	dnsNames := []string{
-		commonPrefix + "." + zoneNames[0] + "." + regionName + "." + serviceDnsSuffix, // zone level - TODO might need other zone names for multi-zone clusters
-		commonPrefix + "." + regionName + "." + serviceDnsSuffix,                      // region level, one up from zone level
-		commonPrefix + "." + serviceDnsSuffix,                                         // global level, one up from region level
-		"", // nowhere to go up from global level
-	}
 
-	endpoints := [][]string{zoneEndpoints, regionEndpoints, globalEndpoints}
+	return nil
+}
 
-	dnsZone, err := getDnsZone(s.zoneName, s.zoneID, s.dnsZones)
-	if err != nil {
-		return err
+// reconcileDNS updates the dns records for federated service in the configured dns provider.
+func (dc *DNSController) reconcileDNS(service *apiv1.Service) {
```

Comment:
I would suggest staying with the Kubernetes-wide convention of "ensure" for all anti-entropy operations, rather than adding a new term "reconcile".